Dockerfile

```diff
@@ -1,33 +1,34 @@
-FROM alpine:3.4
-MAINTAINER Thomas Boerger <thomas@webhippie.de>
+#Build stage
+FROM sapk/alpine-multi AS build-env
+
+ARG TAGS="sqlite"
+ENV TAGS "bindata ${TAGS}"
+ENV GOPATH /go/
+ENV PATH "${PATH}:${GOPATH}/bin"
+
+RUN apk --no-cache add go git build-base
+
+ADD . ${GOPATH}/src/code.gitea.io/gitea
+WORKDIR ${GOPATH}/src/code.gitea.io/gitea
+
+ARG GITEA_VERSION
+RUN if [ -n "${GITEA_VERSION}" ]; then git checkout ${GITEA_VERSION}; fi \
+ && make clean generate build
+
+#Run stage
+FROM sapk/alpine-multi
+LABEL maintainer "Thomas Boerger <thomas@webhippie.de>"
+
+ARG UID=1000
+ARG GID=1000
 
 EXPOSE 22 3000
 
-RUN apk update && \
-  apk add \
-    su-exec \
-    ca-certificates \
-    sqlite \
-    bash \
-    git \
-    linux-pam \
-    s6 \
-    curl \
-    openssh \
-    tzdata && \
-  rm -rf \
-    /var/cache/apk/* && \
-  addgroup \
-    -S -g 1000 \
-    git && \
-  adduser \
-    -S -H -D \
-    -h /data/git \
-    -s /bin/bash \
-    -u 1000 \
-    -G git \
-    git && \
-  echo "git:$(date +%s | sha256sum | base64 | head -c 32)" | chpasswd
+RUN apk --no-cache add \
+    su-exec ca-certificates sqlite bash git linux-pam s6 curl openssh tzdata \
+ && addgroup -S -g ${GID} git \
+ && adduser -S -H -D -h /data/git -s /bin/bash -u ${UID} -G git git \
+ && echo "git:$(dd if=/dev/urandom bs=24 count=1 | base64)" | chpasswd
 
 ENV USER git
```

Comment:
Reminder to myself : Check if this is usefull since this still run with root after.